src/visitors/rules/custom/regRule.js

```diff
@@ -0,0 +1,75 @@
+/* eslint-disable no-underscore-dangle */
+
+class RegRule {
+  static get rule() {
+    return 'reg-rule';
+  }
+
+  constructor(config) {
+    this.ruleSets = {};
+
+    for (const ruleName of Object.keys(config.rules || {})) {
+      const [treeLevel, field] = config.rules[ruleName].on.split('.');
+      const rule = {
+        not: false,
+        ...config.rules[ruleName],
+        treeLevel,
+        field,
+        name: ruleName,
+      };
+
+      const regexp = (!!rule.regexp && new RegExp(rule.regexp)) || null;
+      const startsWith = (!!rule.startsWith && new RegExp(`^${rule.startsWith}`)) || null;
```

Comment:
I would better check startsWith and endsWith with function vs regexp. Or we need to escape e.g. `.` and other symbols here (see regexp based discussion above)